packages/mui-material/src/Select/SelectInput.js

```diff
@@ -324,6 +328,32 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     }
   };
 
+  const handleItemKeyDown = (event) => {
+    if (event.key === ' ') {
+      const child = childrenArray.find(
+        (childItem) => childItem.props.value === event.target.dataset.value,
+      );
```

Comment:
Should be handled like `onClick: handleItemClick(child),` no?

Reply:
Yes, this is much better. updated in this commit https://github.com/mui/material-ui/pull/43966/commits/9e01896f595f6b8ea76d8b0bda54c17e806a9749